vpr/src/route/route_timing.cpp

```diff
@@ -1328,7 +1420,7 @@ static t_rt_node* setup_routing_resources(int itry,
 
         // check for edge correctness
         VTR_ASSERT_SAFE(is_valid_skeleton_tree(rt_root));
-        VTR_ASSERT_SAFE(should_route_net(net_id, connections_inf, true));
+        VTR_ASSERT_SAFE(should_route_net(net_id, connections_inf, true) || init_path_data_struct);
```

Comment:
Can you explain why this change was made?

Reply:
This is to prevent VPR from throwing an exception when a net is told to reroute by RCV, as opposed to the `should_route_net` function. RCV tells the router to reroute nets if the path cost is below the minimum budget, but it doesn't use the `should_route_net` function, it uses a function in the route_budgets class, `get_should_reroute`. As a result this assertion can fail.

Reply:
What you wrote in reply doesn't match how the code reads.  Can the variables be better named, or add a comment?

Reply:
I agree the variable definitely can be renamed!